api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -297,9 +298,28 @@ message ClusterManager {
 // Envoy process watchdog configuration. When configured, this monitors for
 // nonresponsive threads and kills the process after the configured thresholds.
 // See the :ref:`watchdog documentation <operations_performance_watchdog>` for more information.
+// [#next-free-field: 8]
 message Watchdog {
   option (udpa.annotations.versioning).previous_message_type = "envoy.config.bootstrap.v2.Watchdog";
 
+  message WatchdogAction {
+    // In priority events are fired from KILL, MULTIKILL, MEGAMISS and MISS.
+    enum WatchdogEvent {
+      UNKNOWN = 0;
+      KILL = 1;
+      MULTIKILL = 2;
+      MEGAMISS = 3;
+      MISS = 4;
+    }
+
+    // The name of the action. This is just a well-known string that listeners can
+    // use for registering callbacks. Custom actions should be named using reverse
+    // DNS to ensure uniqueness.
+    string name = 1;
+
+    WatchdogEvent event = 2;
+  }
```

Comment:
This makes me thing of overload manager. I wonder if long term we just want a notion of registered callbacks and events, which isn't tied to the specific of watchdog or overload manager, etc.